diff --git a/includes/registration/ExtensionProcessor.php b/includes/registration/ExtensionProcessor.php
index 8726f3ddde..dade2ea196 100644
--- a/includes/registration/ExtensionProcessor.php
+++ b/includes/registration/ExtensionProcessor.php
@@ -66,6 +66,7 @@ class ExtensionProcessor implements Processor {
 		'ParsoidModules',
 		'RestRoutes',
 		'SkinOOUIThemes',
+		'SkinOOUIThemePaths',
 		'SearchMappings',
 		'TrackingCategories',
 	];
